### ISSUE TITLE
Secondary search

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#118128]
I'd like to be able to search for something, for example `ipad` and then in the files which contain that term, search for something else, for example `terminal`.

IE find files which contain ____ && ____.

**Comment by roblourens** [comment]
Do you want to search in the line with the match, and the surrounding lines, or anywhere in the file? If the first, try Search Editors.

**Comment by meganrogge** [comment]
Anywhere in the file

**Comment by meganrogge** [comment]
It would've saved me so much time to have this capability on 3 unique occasions today. Talked w @kieferrm and I hope to work on a PR for ripgrep in the coming iterations so that we have a:

SearchTerm1 && SearchTerm2 option

**Comment by roblourens** [comment]
Could also be a good extension candidate using the (proposed) `findTextInFiles` API. 

For your original case, it sounds like what I often use "files to include" for, when I want to search for the use of a term like ipad in some feature area like notebooks, I would use `*Notebook*` (note that it's case sensitive)

**Comment by ArturoDent** [comment]
I made this extension which works around getting the current search results, regex-ing for the filenames, getting their relative paths and then putting those paths into the 'files to include` filter.  [Find and Transform](https://marketplace.visualstudio.com/items?itemName=ArturoDent.find-and-transform&ssr=false#overview).

Specifically [Using the Search Results files in another search](https://github.com/ArturoDent/find-and-transform/blob/HEAD/searchInPanel.md#using-the-search-results-files-in-another-search).

Use from the context menu, Command Palette or a keybinding like:

```jsonc
{
  "key": "shift+alt+s",  
  "command": "find-and-transform.searchInResults",
  "args": {
    "find": "Second",
    "replace": "Fourth",
    "filesToInclude": "${file}",  // will be ignored in this command, ${resultsFiles} automatically applied
    "triggerSearch": false        // will be ignored, true will be applied

		// other available args

		// "triggerReplaceAll"
		// "isRegex"
		// "filesToInclude"
		// "preserveCase"
		// "useExcludeSettingsAndIgnoreFiles"
		// "isCaseSensitive"
		// "matchWholeWord"
		// "filesToExclude"
  },
  "when": "hasSearchResult"      // I suggest this but it isn't mandatory
}
```
There are other ways to use the `${resultsFiles}` variabled as well.

**Comment by osortega** [comment]
This is a duplicate of https://github.com/microsoft/vscode/issues/197634
Closing

/duplicate

**Comment by gjsjohnmurray** [comment]
@osortega I disagree about this being a duplicate. That one is about tree filtering. And anyway it's locked so unavailable for discussion.
